api/environmentvariables/env_vars_handler.go

```diff
@@ -0,0 +1,176 @@
+package environmentvariables
+
+import (
+	"fmt"
+	"github.com/equinor/radix-operator/pkg/apis/deployment"
+	"sort"
+	"strings"
+
+	"github.com/equinor/radix-api/api/deployments"
+	envvarsmodels "github.com/equinor/radix-api/api/environmentvariables/models"
+	"github.com/equinor/radix-api/api/events"
+	"github.com/equinor/radix-api/models"
+	"github.com/equinor/radix-operator/pkg/apis/kube"
+	v1 "github.com/equinor/radix-operator/pkg/apis/radix/v1"
+	crdUtils "github.com/equinor/radix-operator/pkg/apis/utils"
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// EnvVarsHandlerOptions defines a configuration function
+type EnvVarsHandlerOptions func(*EnvVarsHandler)
+
+// WithAccounts configures all EnvVarsHandler fields
+func WithAccounts(accounts models.Accounts) EnvVarsHandlerOptions {
+	return func(eh *EnvVarsHandler) {
+		kubeUtil, _ := kube.New(accounts.UserAccount.Client, accounts.UserAccount.RadixClient)
+		eh.kubeUtil = *kubeUtil
+		eh.inClusterClient = accounts.ServiceAccount.Client
+		eh.deployHandler = deployments.Init(accounts)
+		eh.eventHandler = events.Init(accounts.UserAccount.Client)
+		eh.accounts = accounts
+	}
+}
+
+// WithEventHandler configures the eventHandler used by EnvVarHandler
+func WithEventHandler(eventHandler events.EventHandler) EnvVarsHandlerOptions {
+	return func(eh *EnvVarsHandler) {
+		eh.eventHandler = eventHandler
+	}
+}
+
+// EnvVarsHandler Instance variables
+type EnvVarsHandler struct {
+	kubeUtil        kube.Kube
+	inClusterClient kubernetes.Interface
+	deployHandler   deployments.DeployHandler
+	eventHandler    events.EventHandler
+	accounts        models.Accounts
+}
+
+// Init Constructor.
+// Use the WithAccounts configuration function to configure a 'ready to use' EnvVarsHandler.
+// EnvVarsHandlerOptions are processed in the seqeunce they are passed to this function.
+func Init(opts ...EnvVarsHandlerOptions) EnvVarsHandler {
+	eh := EnvVarsHandler{}
+
+	for _, opt := range opts {
+		opt(&eh)
+	}
+
+	return eh
+}
+
+//GetComponentEnvVars Get environment variables with metadata for the component
+func (eh EnvVarsHandler) GetComponentEnvVars(appName string, envName string, componentName string) ([]envvarsmodels.EnvVar, error) {
+	apiEnvVars, _, _, _, err := eh.getComponentEnvVars(appName, envName, componentName)
+	if err != nil {
+		return nil, err
+	}
+	sort.Slice(apiEnvVars, func(i, j int) bool { return apiEnvVars[i].Name < apiEnvVars[j].Name })
+	return apiEnvVars, nil
+}
+
+//ChangeEnvVar Change environment variables
+func (eh EnvVarsHandler) ChangeEnvVar(appName, envName, componentName string, envVars []envvarsmodels.EnvVarParameter) error {
+	apiEnvVars, currentEnvVarsConfigMap, currentEnvVarsMetadataConfigMap, envVarsMetadataMap, err := eh.getComponentEnvVars(appName, envName, componentName)
```

Comment:
We should find a bullet proof way to only allow editing variables defined in radixconfig, and not system generated RADIX-*.
It works now because adding new variables is not supported as variable names are checked against apiEnvVarsMap. But apiEnvVarsMap also contain RADIX_* variables, so if we are not careful, we could end up adding RADIX_* variables to the configmap.

Reply:
Created a method `kube.IsRadixEnvVar(envVarParam.Name)` -(in `radix-operator`). I'm not sure if it is bulletproof though.